app/controllers/concerns/authentication.rb

```diff
@@ -47,13 +52,24 @@ def user
 
   def rbac_allowed?
     return true if ActiveModel::Type::Boolean.new.cast(Settings.disable_rbac)
+    return true if skip_rbac_for_cert_based_auth?
 
     @rbac_api ||= ::RbacApi.new(request.headers['X-RH-IDENTITY'])
     @rbac_api.check_user
   end
 
   private
 
+  def skip_rbac_for_cert_based_auth?
+    cert_based_auth? &&
+      ALLOWED_CERT_BASED_RBAC_ACTIONS.include?(controller: controller_name,
+                                               action: action_name)
+  end
+
+  def cert_based_auth?
+    identity_header.identity.dig('user').nil?
```

Comment:
Seems rather weak, but since the identity_header is something we can only get from 3scale, I guess it's fine